cogs/channel.py

```diff
@@ -0,0 +1,144 @@
+import csv
+import yaml
+from discord.ext import commands
+from discord.ext.commands import has_permissions
+from discord.utils import find
+import re
+
+if "CompsciBot" not in str(os.getcwd()):
+    os.chdir("./CompsciBot")
+with open("config.yaml") as file:
+    config = yaml.load(file, Loader=yaml.FullLoader)
+
+# Here we name the cog and create a new class for the cog.
+testGuild = 931309621027684413
+
+
+class ChannelManager(commands.Cog, name="channelmanager"):
+    def __init__(self, bot):
+        self.bot = bot
+
+    async def createCategory(category, context):
+        """
+        creates a category
+
+            Parameters:
+                category: the name of the category
+                context: the context of the command
+
+            Returns:
+                the category object
+        """
+
+        guild = context.guild
+
+        try:
+            return await guild.create_category(category)
+        except:
+            print("Issue with ", category, ".  Error: ", sys.exc_info()[0])
+
+    async def getCategory(category_name, context):
+        """
+        Finds a category by same or if none exists creates a new one
+            Parameters:
+                category: the name of the category
+                context: the context of the command
+
+            Returns:
+                the category object
+        """
+        guild = context.guild
+        category = find(lambda category: category.name ==
+                        category_name, guild.categories)
+        # get category by category name
+        if(category == None):
+            category = await guild.create_category(category_name)
+        return category
+
+    async def createChannel(channel_name: str, category_name: str, context):
+        """
+        Given a channel and a category name, creates a channel and assigns it to the category
+            Parameters:
+                channel_name: the name of the channel
+                category_name: the name of the category
+        """
+        guild = context.guild
+
+        category = await ChannelManager.getCategory(category_name, context)
+
+        return await guild.create_text_channel(channel_name, category=category)
+
+    def makeDict(categories, channels):
+        classDict = {}
+        for channel in channels:
+            for category in categories:
+                if (category in channel):
+                    classDict[channel] = category
+        print(classDict)
+        return classDict
+
+    @commands.command(name="channelparse")
+    @has_permissions(administrator=True)
+    async def channelparse(self, context):
+        """
+        [Optional filename argument] This command sets up channels from a csv
+        """
+
+        channelnames = []
+        categories = []
+
+        # parses command to get message content
+        prefix = config["bot_prefix"]
+        startLen = len(prefix) + len("channelparse")
+        filename = context.message.content[startLen:].strip()
```

Comment:
I don't think you have to do this to get the filename, look at how the TLDR method gets the URL, see if you can do something similar: https://github.com/Nanosplitter/CompsciBot/blob/4fe1f6531e4be9104cc16c8822a76883ef12a05b/cogs/general.py#L146